src/search/parser_tests.h

```diff
@@ -0,0 +1,210 @@
+        indent += "    ";
+        std::cout << indent << "reopen_closed = " << reopen_closed << std::endl;
+        std::cout << indent << "f_eval = " << std::endl;
+        if (f_eval) {
+            f_eval->dump(indent + "    ");
+        } else {
+            std::cout << indent << "    " << "Not specified on command line" << std::endl;
+        }
+        std::cout << indent << "preferred = " << std::endl;
+        for (const auto& h : preferred) {
+            h->dump(indent + "    ");
+        }
+    }
+};
+
+class EagerSearchPlugin : public parser::Plugin<SearchEngine, EagerSearch> {
+public:
+    EagerSearchPlugin()
+        : parser::Plugin<SearchEngine, EagerSearch>("eager", "Eager Search") {;
+        document_synopsis("Eager search synopsis");
+
+        add_option<bool>("reopen_closed", "reopen closed nodes", "false");
+        add_option<std::shared_ptr<Evaluator>>(
+            "f_eval",
+            "set evaluator for jump statistics. "
+            "(Optional; if no evaluator is used, jump statistics will not be displayed.)",
+            parser::ArgumentInfo::NO_DEFAULT);
+        add_list_option<std::shared_ptr<Evaluator>>(
+            "preferred",
+            "use preferred operators of these evaluators", "[]");
+    }
+};
+
+class IteratedSearch : public SearchEngine {
+    std::vector<std::unique_ptr<parser::EnrichedASTNode>> searches;
+public:
+    explicit IteratedSearch(const parser::Options &opts)
+        : SearchEngine(opts),
+          searches() {
+        std::shared_ptr<parser::EnrichedASTNode> searches_AST =
+            opts.get<std::shared_ptr<parser::EnrichedASTNode>>("searches");
+        std::shared_ptr<parser::EnrichedListNode> searches_AST_list =
+                std::dynamic_pointer_cast<parser::EnrichedListNode>(searches_AST);
+        for (const auto & element : searches_AST_list->get_elements()) {
+            searches.push_back(element->clone());
+        }
+    }
+    virtual ~IteratedSearch() = default;
+    void dump(std::string indent) override {
+        std::cout << indent << "IteratedSearch with " << std::endl;
+        indent += "    ";
+        std::cout << indent << "searches = " << std::endl;
+        for (const auto& search_AST : searches) {
+            parser::Any search_any = search_AST->construct();
+            std::shared_ptr<SearchEngine> search = parser::any_cast<std::shared_ptr<SearchEngine>>(search_any);
+            search->dump(indent + "    ");
+        }
+    }
+};
+
+class IteratedSearchPlugin : public parser::Plugin<SearchEngine, IteratedSearch> {
+public:
+    IteratedSearchPlugin()
+        : parser::Plugin<SearchEngine, IteratedSearch>("iterated", "Iterated Search") {;
+        add_option<std::vector<std::shared_ptr<SearchEngine>>>(
+            "searches",
+            "use preferred operators of these evaluators", "[]", parser::Bounds::unlimited(), true);
+    }
+};
+
+class BlindHeuristicPlugin : public parser::Plugin<Evaluator, BlindHeuristic> {
+public:
```

Comment:
I see the border between what we can do now and what we'll do later like this: the plugin class should more or less reach its final state in this issue but would still create the classes directly (i.e., the `BlindHeuristicPlugin` creates a `BlindHeuristic`). When we work on the component interaction problem, we'll introduce "builders" and the plugins will create a builder instead of creating the class directly. Maybe we can make the builder and the plugin the same then, but I think we need to keep them separate because the "builder" needs to know the task, while the plugin does not. So to answer your question, I would like to get the plugin class right in all aspects except for the ones that relate to replacing `BlindHeuristic` with `BlindHeuristicBuilder`.

Reply:
OK, that's ambitious, but I understand that we don't want to touch all plug-ins many times. In that case we should look into things like avoiding boilerplate.